src/webui/api/appcontroller.cpp

```diff
@@ -1083,6 +1085,22 @@ void AppController::defaultSavePathAction()
     setResult(BitTorrent::Session::instance()->savePath().toString());
 }
 
+void AppController::getDirectoriesAction()
+{
+    requireParams({u"path"_s});
+
+    const QString path = params().value(u"path"_s);
+    if (path.isEmpty())
+        throw APIError(APIErrorType::BadParams);
+
+    QDir dir(path);
+    if(!dir.exists() || !dir.isAbsolute())
```

Comment:
I would consider non-absolute path as "bad params".

Reply:
It seems reasonable